app/services/user_creation_service.rb

```diff
@@ -0,0 +1,45 @@
+# dead simple logger. Expect to see a fake version that's cleverer in specs
+class SimpleLogger
+  def self.log(message)
+    puts message
+  end
+end
+
+# This is here to create users from a rake task.
+class UserCreationService
+  def initialize(params, logger=SimpleLogger, randomizer=SecureRandom)
+    @params = params
+    @logger = logger
+    @randomizer = randomizer
+  end
+
+  def save
+    saved = user.save
+    if saved
+      logger.log("Api User Created")
+      logger.log("================")
+      logger.log("API KEY: #{user.api_key}")
+      logger.log("API SECRET: #{unencrypted_api_secret}")
+      logger.log("BE SURE TO STORE THE SECRET SECURELY")
+      logger.log("WE DO NOT STORE UNENCRYPTED SECRETS")
+      logger.log("WRITE IT DOWN NOW")
+    else
+      logger.log("USER WAS INVALID:")
+      logger.log("ERRORS:")
+      logger.log("#{user.errors.full_messages.join('\n')}")
+    end
+    saved
+  end
+
+  def user
```

Comment:
hmm, I think I had left that in there as a hook for testing, but yeah, in a unit test I could just grab `User.last` and be done with it. I'll make it private.